### PR TITLE
Issue 7031

### PR DESCRIPTION
**Pull request by Yoric**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7032)

### CODE REVIEW

**Comment by frewsxcv** [comment]
@Yoric Did you mean for this to include your commit #7028?

**Comment by Yoric** [comment]
@frewsxcv No, my bad. I'll fix that.

File: tests/reftest.rs [hunk 1]
```diff
@@ -43,7 +43,8 @@ bitflags!(
 
 fn main() {
     let args: Vec<String> = env::args().collect();
-    let mut parts = args.tail().split(|e| &**e == "--");
+
+    let mut parts = args.split_first().unwrap().1.split(|e| &**e == "--");
```
Comment (SimonSapin) [inline tests/reftest.rs]: Nit: I think `args.split_first().unwrap().1` is better written as `args[1..]`. r=me with this.

**Comment by Yoric** [comment]
That's nicer, indeed.

**Comment by jdm** [comment]
Looks like the heap_api feature is still necessary:
```
   Compiling util_tests v0.0.1 (file:///home/servo/buildbot/slave/linux1/build/components/servo)
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:37:33: 37:55 error: use of unstable library feature 'heap_api': the precise API and guarantees it provides may be tweaked slightly, especially to possibly take into account the types being stored to make room for a future tracing garbage collector
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:37         assert_eq!(heap_size_of(::std::rt::heap::EMPTY as *const c_void), 0);
                                                                                                  ^~~~~~~~~~~~~~~~~~~~~~
<std macros>:1:1: 9:39 note: in expansion of assert_eq!
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:37:9: 37:78 note: expansion site
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:37:33: 37:55 help: add #![feature(heap_api)] to the crate attributes to enable
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:40:17: 40:42 error: use of unstable library feature 'heap_api': the precise API and guarantees it provides may be tweaked slightly, especially to possibly take into account the types being stored to make room for a future tracing garbage collector
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:40         let x = ::std::rt::heap::allocate(64, 0);
                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:40:17: 40:42 help: add #![feature(heap_api)] to the crate attributes to enable
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:42:9: 42:36 error: use of unstable library feature 'heap_api': the precise API and guarantees it provides may be tweaked slightly, especially to possibly take into account the types being stored to make room for a future tracing garbage collector
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:42         ::std::rt::heap::deallocate(x, 64, 0);
                                                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:42:9: 42:36 help: add #![feature(heap_api)] to the crate attributes to enable
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:45:17: 45:42 error: use of unstable library feature 'heap_api': the precise API and guarantees it provides may be tweaked slightly, especially to possibly take into account the types being stored to make room for a future tracing garbage collector
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:45         let x = ::std::rt::heap::allocate(255, 0);
                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:45:17: 45:42 help: add #![feature(heap_api)] to the crate attributes to enable
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:47:9: 47:36 error: use of unstable library feature 'heap_api': the precise API and guarantees it provides may be tweaked slightly, especially to possibly take into account the types being stored to make room for a future tracing garbage collector
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:47         ::std::rt::heap::deallocate(x, 255, 0);
                                                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:47:9: 47:36 help: add #![feature(heap_api)] to the crate attributes to enable
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:50:17: 50:42 error: use of unstable library feature 'heap_api': the precise API and guarantees it provides may be tweaked slightly, especially to possibly take into account the types being stored to make room for a future tracing garbage collector
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:50         let x = ::std::rt::heap::allocate(1024 * 1024, 0);
                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:50:17: 50:42 help: add #![feature(heap_api)] to the crate attributes to enable
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:52:9: 52:36 error: use of unstable library feature 'heap_api': the precise API and guarantees it provides may be tweaked slightly, especially to possibly take into account the types being stored to make room for a future tracing garbage collector
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:52         ::std::rt::heap::deallocate(x, 1024 * 1024, 0);
                                                                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/servo/buildbot/slave/linux1/build/tests/unit/util/mem.rs:52:9: 52:36 help: add #![feature(heap_api)] to the crate attributes to enable
error: aborting due to 7 previous errors
Could not compile `util_tests`.
```

**Comment by SimonSapin** [comment]
@Yoric Rather than removing `heap_api`, replace that line and the next with:
```rust
#![cfg_attr(test, feature(plugin, custom_derive, heap_api))]
#![cfg_attr(test, plugin(plugins))]
```

What’s going on is that the unit tests crates are `dev-dependencies` of the Servo crate so that we can test them with `cargo test -p foo` while sharing the build directory. Because of that, these crates are also built as "dependencies" of the reftest harness but with `cfg(test)` disabled so the modules are not compiled and `heap_api` is not used.

But `heap_api` *is* used when the unit test crates is compiled for testing.

**Comment by Yoric** [comment]
Applied feedback.

**Pull request merged**